### ISSUE TITLE
Describe how detectron2 should be added as a dependency

### ISSUE THREAD

**Issue by prhbrt** [facebookresearch/detectron2#2820]
Related to issue #969, which requests submission to pypi.

## 🚀 Feature
Describe a way to or provide a function to add `detectron2` as a dependency.

## Motivation & Examples

I like to publish reusable and reproducable results with the Mask-RCNN, and it would be nice if `detectron2` could install itself by figuring out which `cuda` version is available or complain when there is none. I'm using this in Google Colab notebooks to mimic the behavior, bit would recon the `detectron2` developers could do a significantly better job. Moreover, this script is likely to break at some point when new versions are released. It's not elegant to point users interested in ones research to documentation on how to match `cuda`, `torch`, `torchvision` and `detectron2` versions.

    import torch
    try:
        import detectron2
    except ImportError:
        torch_version, cuda_version = torch.__version__.split('+cu')
        torch_version = '.'.join(torch_version.split('.')[:2])
        !pip3 install -U pyyaml
        !pip3 install detectron2 -f "https://dl.fbaipublicfiles.com/detectron2/wheels/cu{cuda_version}/torch{torch_version}/index.html"
        import detectron2

PS: I, however, enjoy your library in many other ways! Thanks for sharing.

**Comment by ppwwyyxx** [comment]
The script you provided is fine - I don't know if there are any better ways. 

Pytorch3d has adopted similar approach: https://github.com/facebookresearch/pytorch3d/blob/master/INSTALL.md#3-install-wheels-for-linux Would you like to add such a section in our doc as well?
